File: frontend/src/services/api/analytics.ts
import { apiClient } from "../../lib/apiClient";
import { AnalyticsOverviewResponse } from "./types";

/**
 * Get analytics overview for the current user
 */
export async function getAnalyticsOverview(): Promise<AnalyticsOverviewResponse> {
    console.log(
        "%c📊 BIGQUERY ANALYTICS",
        "background: #3B82F6; color: white; font-weight: bold; padding: 4px 8px; border-radius: 4px;",
        "\n📍 Fetching analytics from /api/analytics/overview"
    );

    const { data } = await apiClient.get<AnalyticsOverviewResponse>("/api/analytics/overview");

    console.log(
        "%c✅ BIGQUERY DATA RECEIVED",
        "background: #10B981; color: white; font-weight: bold; padding: 4px 8px; border-radius: 4px;",
        "\n📊 Stats:",
        {
            notes: data.overview.total_notes,
            flashcards: data.overview.total_flashcards,
            reviews: data.overview.total_reviews,
            mastery: `${data.overview.mastery_rate_percent.toFixed(1)}%`,
            activityDays: data.activity.length,
        }
    );

    return data;
}

/**
 * Get flashcard difficulty statistics
 */
export async function getFlashcardDifficulty(limit: number = 20): Promise<any> {
    console.log(
        "%c📊 BIGQUERY ANALYTICS",
        "background: #3B82F6; color: white; font-weight: bold; padding: 4px 8px; border-radius: 4px;",
        `\n📍 Fetching difficulty stats (limit: ${limit})`
    );

    const { data } = await apiClient.get<any>(`/api/analytics/flashcard-difficulty?limit=${limit}`);

    console.log(
        "%c✅ BIGQUERY DATA RECEIVED",
        "background: #10B981; color: white; font-weight: bold; padding: 4px 8px; border-radius: 4px;",
        `\n📊 Flashcards: ${data.flashcards?.length || 0}`
    );

    return data;
}

File: backend-fastapi/app/api/analytics.py
"""Analytics API endpoints"""
from fastapi import APIRouter, Depends
from app.core.auth import get_current_user_id
from app.services.analytics_service import AnalyticsService
from app.models.analytics import AnalyticsOverviewResponse

router = APIRouter(prefix="/api/analytics", tags=["analytics"])


@router.get("/overview", response_model=AnalyticsOverviewResponse)
async def get_analytics_overview(
    user_id: str = Depends(get_current_user_id)
):
    """
    Get analytics overview for the current user.
    
    Returns study statistics from BigQuery including:
    - Total notes, flashcards, and reviews
    - Average interval and mastery rate
    - Daily review activity for the last 30 days
    """
    print("\n" + "🔵" * 40)
    print("📊 API REQUEST: /api/analytics/overview")
    print(f"  Requesting analytics for user: {user_id}")
    print("🔵" * 40 + "\n")
    
    service = AnalyticsService()
    data = await service.get_user_overview(user_id)
    
    print("🟢" * 40)
    print("✅ API RESPONSE: Analytics data ready")
    print("🟢" * 40 + "\n")
    
    return data


@router.get("/flashcard-difficulty")
async def get_flashcard_difficulty(
    limit: int = 20,
    user_id: str = Depends(get_current_user_id)
):
    """
    Get flashcard difficulty statistics.
    
    Returns the most difficult flashcards based on review patterns.
    """
    service = AnalyticsService()
    stats = await service.get_flashcard_difficulty_stats(user_id, limit)
    return {"flashcards": stats}
